src/Agent/NewRelic/Agent/Core/DistributedTracing/DistributedTracePayloadHandler.cs

```diff
@@ -78,7 +78,7 @@ public void InsertDistributedTraceHeaders<T>(IInternalTransaction transaction, T
                 }
                 else
                 {
-                    transaction.SetSampled(_sampler);
+                    transaction.SetSampled(_samplerService.GetSampler(SamplerType.Root)); // TODO: Is Root correct here?
```

Comment:
I _think_ the way you're using it throughout is correct. While we don't strictly know if we want the "root" sampler here, if we didn't, the Sampled value would already be set, and this would do nothing.